manta-crypto/src/arkworks/serialize.rs

```diff
@@ -25,6 +28,153 @@ use {
 #[doc(inline)]
 pub use ark_serialize::*;
 
+/// Serialization Hook
```

Comment:
Can you expand a little on the docs here? I don't understand what "serialization hook" means nor when/how to use this trait.

Reply:
It's a way to prove to the compiler that there exists a serialization implementation for the type you implement this trait on.

Reply:
It's used whenever a `CanonicalSerialize` implementation is missing for a type we don't have control over and so we use our own type and use this hook to get the serialization pushed into the type we don't own.